molgenis-ontology/src/main/java/org/molgenis/ontology/controller/SortaServiceController.java

```diff
@@ -0,0 +1,614 @@
+	public static final String URI = MolgenisPluginController.PLUGIN_URI_PREFIX + ID;
+	private static final String ILLEGAL_PATTERN = "[^0-9a-zA-Z_]";
+	private static final String ILLEGAL_PATTERN_REPLACEMENT = "_";
+	private static final double DEFAULT_THRESHOLD = 100.0;
+
+	@Autowired
+	public SortaServiceController(OntologyService ontologyService, SortaService sortaService,
+			MatchQualityRocService matchQualityRocService, SortaJobFactory sortaMatchJobFactory,
+			ExecutorService taskExecutor, UserAccountService userAccountService, FileStore fileStore,
+			MolgenisPermissionService molgenisPermissionService, DataService dataService,
+			LanguageService languageService, MenuReaderService menuReaderService)
+	{
+		super(URI);
+		this.ontologyService = requireNonNull(ontologyService);
+		this.sortaService = requireNonNull(sortaService);
+		this.matchQualityRocService = requireNonNull(matchQualityRocService);
+		this.sortaMatchJobFactory = requireNonNull(sortaMatchJobFactory);
+		this.taskExecutor = requireNonNull(taskExecutor);
+		this.userAccountService = requireNonNull(userAccountService);
+		this.fileStore = requireNonNull(fileStore);
+		this.molgenisPermissionService = requireNonNull(molgenisPermissionService);
+		this.dataService = requireNonNull(dataService);
+		this.languageService = requireNonNull(languageService);
+		this.menuReaderService = requireNonNull(menuReaderService);
+	}
+
+	@RequestMapping(method = GET)
+	public String init(Model model)
+	{
+		model.addAttribute("existingTasks", getJobsForCurrentUser());
+		return VIEW_NAME;
+	}
+
+	@RequestMapping(method = GET, value = "/jobs")
+	@ResponseBody
+	public List<Entity> getJobs(Model model)
+	{
+		return getJobsForCurrentUser();
+	}
+
+	@RequestMapping(method = GET, value = "/newtask")
+	public String matchTask(Model model)
+	{
+		model.addAttribute("ontologies", ontologyService.getOntologies());
+		return VIEW_NAME;
+	}
+
+	@RequestMapping(method = RequestMethod.GET, value = "/calculate/{entityName}")
+	public String calculateRoc(@PathVariable String entityName, Model model)
+			throws IOException, MolgenisInvalidFormatException
+	{
+		model.addAllAttributes(matchQualityRocService.calculateROC(entityName));
+		return init(model);
+	}
+
+	@RequestMapping(method = POST, value = "/threshold/{entityName}")
+	public String updateThreshold(@RequestParam(value = "threshold", required = true) String threshold,
+			@PathVariable String entityName, Model model)
+	{
+		if (!StringUtils.isEmpty(threshold))
+		{
+			Entity entity = dataService.findOne(MatchingTaskEntityMetaData.ENTITY_NAME,
+					new QueryImpl().eq(MatchingTaskEntityMetaData.IDENTIFIER, entityName));
+			try
+			{
+				Double threshold_value = Double.parseDouble(threshold);
+				entity.set(MatchingTaskEntityMetaData.THRESHOLD, threshold_value);
+				dataService.update(MatchingTaskEntityMetaData.ENTITY_NAME, entity);
+				dataService.getRepository(MatchingTaskEntityMetaData.ENTITY_NAME).flush();
```

Comment:
Yes because the threshold is a field in the MatchTask table. By changing the threshold, users can update the threshold value in the database and then re-render the value to show the matched and unmatched results